src/libs/TransactionUtils.ts

```diff
@@ -0,0 +1,358 @@
+    );
+}
+
+/**
+ * Given the edit made to the money request, return an updated transaction object.
+ */
+function getUpdatedTransaction(transaction: Transaction, transactionChanges: TransactionChanges, isFromExpenseReport: boolean): Transaction {
+    // Only changing the first level fields so no need for deep clone now
+    const updatedTransaction = {...transaction};
+    let shouldStopSmartscan = false;
+
+    // The comment property does not have its modifiedComment counterpart
+    if (Object.hasOwn(transactionChanges, 'comment')) {
+        updatedTransaction.comment = {
+            ...updatedTransaction.comment,
+            comment: transactionChanges.comment,
+        };
+    }
+    if (Object.hasOwn(transactionChanges, 'created')) {
+        updatedTransaction.modifiedCreated = transactionChanges.created;
+        shouldStopSmartscan = true;
+    }
+    if (Object.hasOwn(transactionChanges, 'amount') && typeof transactionChanges.amount === 'number') {
+        updatedTransaction.modifiedAmount = isFromExpenseReport ? -transactionChanges.amount : transactionChanges.amount;
+        shouldStopSmartscan = true;
+    }
+    if (Object.hasOwn(transactionChanges, 'currency')) {
+        updatedTransaction.modifiedCurrency = transactionChanges.currency;
+        shouldStopSmartscan = true;
+    }
+
+    if (Object.hasOwn(transactionChanges, 'merchant')) {
+        updatedTransaction.modifiedMerchant = transactionChanges.merchant;
+        shouldStopSmartscan = true;
+    }
+
+    if (Object.hasOwn(transactionChanges, 'category') && typeof transactionChanges.category === 'string') {
+        updatedTransaction.category = transactionChanges.category;
+    }
+
+    if (shouldStopSmartscan && transaction?.receipt && Object.keys(transaction.receipt).length > 0 && transaction?.receipt?.state !== CONST.IOU.RECEIPT_STATE.OPEN) {
+        updatedTransaction.receipt.state = CONST.IOU.RECEIPT_STATE.OPEN;
+    }
+
+    updatedTransaction.pendingFields = {
+        ...(Object.hasOwn(transactionChanges, 'comment') && {comment: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'created') && {created: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'amount') && {amount: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'currency') && {currency: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'merchant') && {merchant: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'category') && {category: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+    };
+
+    return updatedTransaction;
+}
+
+/**
+ * Retrieve the particular transaction object given its ID.
+ *
+ * @deprecated Use withOnyx() or Onyx.connect() instead
+ */
+function getTransaction(transactionID: string): Transaction | Record<string, never> {
```

Comment:
```suggestion
function getTransaction(transactionID: string): Transaction | Record<string, unknown> {
```

Reply:
@fabioh8010 since `getTransaction` can return a transaction or an empty object, I chose `Record<string, never>` to identify the empty object. If I change it to `Record<string, unknown>`, the function will be able to return object with any fields.